idea/src/org/jetbrains/kotlin/idea/inspections/ConstantConditionIfInspection.kt

```diff
@@ -157,27 +159,59 @@ private fun KtExpression.enumEntry(): KtEnumEntry? {
 }
 
 fun KtExpression.replaceWithBranch(branch: KtExpression, isUsedAsExpression: Boolean, keepBraces: Boolean = false) {
-    val lastExpression = when {
-        branch !is KtBlockExpression -> replaced(branch)
+    val caretModel = findExistingEditor()?.caretModel
+
+    val subjectVariable = (this as? KtWhenExpression)?.subjectVariable?.let { it ->
+        if (it.annotationEntries.isNotEmpty()) return@let it
+        val initializer = it.initializer ?: return@let it
+        val references = ReferencesSearch.search(it, LocalSearchScope(this)).toList()
+        if (references.size > 1) return@let it
```

Comment:
I think it'd be more useful to replace `toList` with `singleOrNull`.

Reply:
I want to return null if references size is 0.